Modules/_asynciomodule.c

```diff
@@ -1060,24 +1083,29 @@ _asyncio_Future_remove_done_callback_impl(FutureObj *self, PyTypeObject *cls,
         return NULL;
     }
 
-    // Beware: PyObject_RichCompareBool below may change fut_callbacks.
-    // See GH-97592.
+    // Beware: An evil PyObject_RichCompareBool could change fut_callbacks
+    // or its items (see https://github.com/python/cpython/issues/97592 or
+    // https://github.com/python/cpython/issues/125789 for details).
     for (i = 0;
          self->fut_callbacks != NULL && i < PyList_GET_SIZE(self->fut_callbacks);
          i++) {
-        int ret;
-        PyObject *item = PyList_GET_ITEM(self->fut_callbacks, i);
-        Py_INCREF(item);
-        ret = PyObject_RichCompareBool(PyTuple_GET_ITEM(item, 0), fn, Py_EQ);
+        PyObject *cb_tup = PyList_GET_ITEM(self->fut_callbacks, i);
+        if (!PyTuple_CheckExact(cb_tup) || PyTuple_GET_SIZE(cb_tup) != 2) {
+            PyErr_SetString(PyExc_RuntimeError, "corrupted callback tuple");
+            goto fail;
+        }
+        Py_INCREF(cb_tup);
+        PyObject *cb = PyTuple_GET_ITEM(cb_tup, 0);
```

Comment:
Same thing here, I don't think it's possible to delete `cb` once the tuple has been incref'd, but if it's somehow possible to replace the first item in the tuple then `cb` will be freed possibly causing a UAF.